Microsoft.Toolkit.Uwp.SampleApp/SamplePages/InAppNotification/InAppNotificationPage.xaml.cs

```diff
@@ -0,0 +1,219 @@
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.SampleApp.SamplePages
+{
+    public sealed partial class InAppNotificationPage : Page, INotifyPropertyChanged
+    {
+        private ControlTemplate _defaultInAppNotificationControlTemplate;
+
+        public bool IsRootGridActualWidthLargerThan700 { get; set; }
+
+        public int NotificationDuration { get; set; } = 0;
+
+        public InAppNotificationPage()
+        {
+            InitializeComponent();
+
+            _defaultInAppNotificationControlTemplate = ExampleInAppNotification.Template;
+        }
+
+        private void SetDefaultControlTemplate()
+        {
+            // Update control template
+            ExampleInAppNotification.Template = _defaultInAppNotificationControlTemplate;
+        }
+
+        private void NotificationDurationTextBox_TextChanged(object sender, TextChangedEventArgs e)
+        {
+            int newDuration;
+            if (int.TryParse(NotificationDurationTextBox.Text, out newDuration))
+            {
+                NotificationDuration = newDuration;
+            }
+        }
+
+        private async void ShowNotificationWithRandomTextButton_Click(object sender, RoutedEventArgs e)
+        {
+            ExampleVSCodeInAppNotification.Dismiss();
+            SetDefaultControlTemplate();
+
+            var random = new Random();
+            int result = random.Next(1, 4);
+
+            if (result == 1)
+            {
+                await ExampleInAppNotification.ShowAsync("Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sollicitudin bibendum enim at tincidunt. Praesent egestas ipsum ligula, nec tincidunt lacus semper non.", NotificationDuration);
+            }
+
+            if (result == 2)
+            {
+                await ExampleInAppNotification.ShowAsync("Pellentesque in risus eget leo rhoncus ultricies nec id ante.", NotificationDuration);
+            }
+
+            if (result == 3)
+            {
+                await ExampleInAppNotification.ShowAsync("Sed quis nisi quis nunc condimentum varius id consectetur metus. Duis mauris sapien, commodo eget erat ac, efficitur iaculis magna. Morbi eu velit nec massa pharetra cursus. Fusce non quam egestas leo finibus interdum eu ac massa. Quisque nec justo leo. Aenean scelerisque placerat ultrices. Sed accumsan lorem at arcu commodo tristique.", NotificationDuration);
+            }
+        }
+
+        private async void ShowNotificationWithButtonsButton_Click(object sender, RoutedEventArgs e)
+        {
+            ExampleVSCodeInAppNotification.Dismiss();
+            SetDefaultControlTemplate();
+
+            var grid = new Grid();
```

Comment:
I wanted to show different examples : so there is one with XAML DataTemplate and another one using C#.